go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -682,16 +694,15 @@ func (app *rootHashApplication) postProcessFinalizedBlock(ctx *abci.Context, rtS
 				logging.LogEvent, roothash.LogEventMessageUnsat,
 			)
 
-			// Roll back changes from message handling.
-			checkpoint.Rollback()
-
 			// Substitute empty block.
 			app.emitEmptyBlock(ctx, rtState, block.RoundFailed)
 
 			return nil
 		}
 	}
 
+	checkpoint.Commit()
```

Comment:
different pattern from elsewhere, which used an `ok` flag and a branch in the `defer` func to either close or commit. this one relies on `Commit(); Close()` being legal

Reply:
Ah that's a good point, I think this is actually a nicer pattern.

> this one relies on Commit(); Close() being legal

Which is.

Reply:
Changed it so this pattern is used everywhere.

Reply:
ok